run.py

```diff
@@ -905,6 +906,99 @@ def hvreads(args):
             "aws", "s3", "cp", output, "%s/%s/hvreads/%s" % (
                S3_BUCKET, args.bioproject, output)])
 
+def alignments(args):
+   available_inputs = get_files(args, "hvreads")
+   existing_outputs = get_files(args, "alignments")
+
+   for sample in get_samples(args):
+      output = "%s.alignments.tsv" % sample
+      if output in existing_outputs: continue
+
+      input_fname = "%s.hvreads.json" % sample
+      if input_fname not in available_inputs: continue
+
+      with tempdir("alignments", input_fname) as workdir:
+         subprocess.check_call([
+            "aws", "s3", "cp", "%s/%s/hvreads/%s" % (
+               S3_BUCKET, args.bioproject, input_fname), input_fname])
+
+         pair1 = []
+         pair2 = []
+         combined = []
+         with open(input_fname) as inf:
+            for seq_id, details in sorted(json.load(inf).items()):
+               if type(details[0]) == type(1):
+                  assignment = details.pop(0)
+               kraken_info, *reads = details
+               assert len(reads) in [1, 2]
+               if len(reads) == 1:
+                  read, = reads
+                  combined.append((seq_id, read))
+               else:
+                  read1, read2 = reads
+                  pair1.append((seq_id, read1))
+                  pair2.append((seq_id, read2))
+
+         for label, reads in [
+               ("pair1", pair1),
+               ("pair2", pair2),
+               ("combined", combined),
+         ]:
+            if reads:
+               with open("%s.fastq" % label, "w") as outf:
+                  for seq_id, read in reads:
+                     seq, quality = read
+                     outf.write("@%s\n%s\n+\n%s\n" % (seq_id, seq, quality))
+
+         sam_out = "out.sam"
+         cmd = [
+            "/home/ec2-user/bowtie2-2.5.2-linux-x86_64/bowtie2",
+            "--local",
+            "-x", "/home/ec2-user/bowtie/human-viruses",
+            "--very-sensitive-local",
+            "--score-min", "G,1,0",
+            "--mp", "2,0",
+            "--no-unal",
+            "--threads", "24",
+            # Including an @SQ line makes the .sam files really big, and we
+            # don't actually need to know the lengths of all the genomes we
+            # considered aligning against. When we have a small number of reads
+            # per sample this can bloat the files 100x.  We'd like to do
+            # "--no-sq" bur pysam chokes on SAM files that are missing the @SQ
+            # header lines.
```

Comment:
So if I understand this correctly... we'd like to do "--no-sq" but we can't, so currently we just have to accept including the "@ SQ" like and making the .sam files bigger than we'd like?

Reply:
That's right!